sklearn/model_selection/_search.py

```diff
@@ -520,6 +548,475 @@ def inverse_transform(self, Xt):
         """
         return self.best_estimator_.transform(Xt)
 
+
+class UnsupervisedSearchMixin(object):
+
+    """Adds unsupervised search to BaseSearch."""
+
+    def _fit(self, X, y=None, parameter_iterable=None):
+        """Draws heavily on the _fit method of BaseSearchCV.
+        """
+        self.scorer_ = check_scoring(self.estimator, scoring=self.scoring)
+        estimator = self.estimator
+
+        grid_scores = list()
+        out = Parallel(n_jobs=1
+                       )(
+            delayed(fit_and_score_unsupervised)(clone(estimator),
+                                                X, scorer=self.scorer_,
+                                                verbose=self.verbose,
+                                                parameters=parameters,
+                                                fit_params=self.fit_params,
+                                                error_score=self.error_score)
+            for parameters in parameter_iterable)
+
+        for score, parameters, _time in out:
+            grid_scores.append(_ScoreTuple(parameters, score))
+
+        # Store the computed scores
+        self.grid_scores_ = grid_scores
+
+        # Find the best parameters by comparing on the score:
+        # note that `sorted` is deterministic in the way it breaks ties
+        best = sorted(grid_scores, key=lambda x: x.score,
+                      reverse=True)[0]
```

Comment:
Why do we need to sort it? Can we not just return the one with the highest score? I understand this is how it was done previously for the regular `GridSearchCV`, but I am unable to figure out the reason. @amueller